src/QAT/qat/purr/backends/verification.py

```diff
@@ -89,25 +101,77 @@ def get_verification_model(qpu_type: QPUVersion) -> Optional[VerificationModel]:
         )
 
     if qpu_type.make == Lucy.__name__:
-        # TODO: Should have an apply_setup_to_hardware in live which people can use to build our own architecture.
-        model = VerificationModel(qpu_type, LucyVerificationEngine)
-        raise NotImplementedError("No lucy-specific model created yet.")
+        return apply_setup_to_hardware(
+            hw= VerificationModel(qpu_type, LucyVerificationEngine),
+            qubit_count=8,
+            pulse_hw_x_pi_2_width=25e-9,
+            pulse_hw_zx_pi_4_width=3.18e-07,
+            pulse_measure_width=2.41e-06,
+        )
 
     return None
 
 
 class VerificationEngine(QuantumExecutionEngine, ABC):
-    def _execute_on_hardware(self, sweep_iterator, package: QatFile):
-        veri_list = list(package.meta_instructions)
-        veri_list.extend(package.instructions)
-        self.verify_instructions(veri_list)
+
+    def _execute_on_hardware(self, sweep_iterator, package: QatFile) -> bool:
+
+        #TODO: Need a discussion around this step. Where are instructions 'finalised'?
+        while not sweep_iterator.is_finished():
+            sweep_iterator.do_sweep(package.instructions)
+            position_map = self.create_duration_timeline(package)
+
+        return self.verify_instructions(position_map)
+
+    def _process_results(self, results, qat_file):
+        return results
+
+    def _process_assigns(self, results, qat_file):
+        return results
 
     @abstractmethod
     def verify_instructions(self, instructions: List[Instruction]):
         ...
 
 
 class LucyVerificationEngine(VerificationEngine):
-    def verify_instructions(self, instructions: List[Instruction]):
-        # TODO: Add actual verification
-        raise NotImplementedError("No lucy-specific verification yet.")
+
+    max_circuit_duration = 90000e-9
+
+    def verify_instructions(self, instructions: dict) -> bool:
+        valid_circuit = True
+
+        if self._get_circuit_duration(instructions) > self.max_circuit_duration:
+            valid_circuit= False
+
+        return valid_circuit
+
+    def _process_results(self, results, qat_file):
+        return results
+
+    def _process_assigns(self, results, qat_file):
+        return results
+
+    def _get_circuit_duration(self, position_map: dict) -> float:
+        pc2samples = {pc: positions[-1].end for pc, positions in position_map.items()}
+        durations = {pc: samples * pc.sample_time for pc, samples in pc2samples.items()}
+
+        circuit_duration = max([duration for duration in durations.values()])
+
+        log.info(
+            f"The circuit duration is {circuit_duration/1e-6} microseconds."
+        )
+
+        return circuit_duration
+
+
+def verify_program(
+    program: str, compiler_config: CompilerConfig, qpu_version: QPUVersion
+):
+    model = get_verification_model(qpu_version)
+
+    return execute(program, model, compiler_config)
+
+
+class VerificationError(Exception):
```

Comment:
This should be thrown if verification fails, otherwise it just dosen't exception.